lib/tasks/application.rake

```diff
@@ -122,4 +124,46 @@ namespace :application do
     Rails.logger = ActiveSupport::Logger.new($stdout)
     Rails.logger.level = 1
   end
+
+  namespace :links_valid do
+    desc "Checks the validity of links in division summary"
+    task divisions: :environment do
+      include PathHelper
+
+      md = Redcarpet::Markdown.new(Redcarpet::Render::HTML)
+      broken_url_constant = 999
+      url_hash_table = Hash.new(0)
+      Division.find_each do |division|
+        wiki_motion = division.wiki_motion
+
+        if wiki_motion
+          summary_in_html = md.render(wiki_motion.description)
+          parsed_data = Nokogiri::HTML.parse(summary_in_html)
+          broken_urls = []
+
+          tags = parsed_data.xpath("//a")
+          tags.each do |tag|
+            url = tag[:href]
+
+            begin
+              if (url_hash_table[url]).zero?
+                url_hash_table[url] += 1
+                uri = URI(url)
+                Net::HTTP.get(uri)
+              elsif url_hash_table[url] == broken_url_constant
+                broken_urls << url
+              end
+            rescue StandardError
+              url_hash_table[url] = broken_url_constant
+              broken_urls << url
+            end
+          end
+          puts "There are broken links in the description for division #{division_url_simple(division)}"
```

Comment:
Won't this get displayed even if there aren't any broken links on a page?

Reply:
Fixed in PR #1318